plugins/Physics/Physics.cpp

```diff
@@ -738,6 +740,73 @@ void Physics::Impl::UpdatePhysics(EntityComponentManager& _ecm)
             return true;
         });
 
+    // Process WorldVelocityCmd
+    _ecm.Each<components::Model, components::WorldVelocityCmd>([&](const Entity& _entity,
+                                                                   const components::Model*,
+                                                                   components::WorldVelocityCmd*
+                                                                       _modelWorldVelocityCmd) {
+        auto modelIt = this->entityModelMap.find(_entity);
+        if (modelIt == this->entityModelMap.end())
+            return true;
+
+        // TODO(addisu) Store the free group instead of searching for it at
+        // every iteration
+        auto freeGroup = modelIt->second->FindFreeGroup();
+
+        // The FreeGroup is created only for floating-base object that do not have any defined
+        // joint between the world and their base
+        if (!freeGroup) {
+            ignwarn << "Failed to find FreeGroup. Linear and angular velocities commands ignored."
+                    << std::endl;
+            return true;
+        }
+
+        // Get canonical link offset
+        auto canonicalLink = _ecm.ChildrenByComponents(_entity, components::CanonicalLink());
+        if (canonicalLink.empty())
+            return true;
+
+        auto canonicalPoseComp = _ecm.Component<components::Pose>(canonicalLink[0]);
+        if (nullptr == canonicalPoseComp)
+            return true;
+
+        ignition::math::Vector3d& modelWorldLinearVelocity = _modelWorldVelocityCmd->Data().linear;
+        ignition::math::Vector3d& modelWorldAngularVelocity =
+            _modelWorldVelocityCmd->Data().angular;
+
+        ignition::math::Pose3d& M_H_B = canonicalPoseComp->Data();
+        ignition::math::Pose3d B_H_M = M_H_B.Inverse();
+        ignition::math::Vector3d& B_o_M = B_H_M.Pos();
+        ignition::math::Quaterniond& B_R_M = B_H_M.Rot();
+
+        {
+            // Express the angular velocity in the frame of the base link
+            ignition::math::Vector3d wBase = B_R_M * modelWorldAngularVelocity;
+            ignwarn << wBase << std::endl;
+
+            // Set the angular velocity of the canonical link
+            freeGroup->SetWorldAngularVelocity(math::eigen3::convert(wBase));
```

Comment:
I thought this method is accepting the velocity of the base expressed with the world orientation (unfortunatly also the ignition physics is quite lacking w.r.t. to this), so the conversion from model to base is unnecessary, if  I am not wrong.

Reply:
The `WorldVelocityCmd` component stores the velocity command of the model frame. Though, this extra frame, is not taken into account by ign-physics. In fact, ign-physics has the concept of [`FreeGroup`](https://bitbucket.org/ignitionrobotics/ign-physics/src/default/include/ignition/physics/FreeGroup.hh) associated to the base link. This means that the value stored in the Cmd components need to be ported to the base link.

In these lines, I get the model velocity expressed in the world frame and, since the model frame and the base link frame belong to the same body, I just rotate it to get the equivalent base link velocity. I didn't get what do you think unnecessary.

Reply:
Sorry for using the notation I am used to, but using words for this kind of stuff is always ambiguous. If you there is a notation that you prefer to use, let me know and I will happilfy use it for my discussions. Following the notation used in https://pure.tue.nl/ws/portalfiles/portal/139293126/A_Multibody_Dynamics_Notation_Revision_2_.pdf, you have that the world ($A$) angular velocity of the frame $B$ and the one of the frame $M$ are already the same vector, see: 
$$
{}^A \omega_{A, B} = {}^A \omega_{A, M} 
$$

Reply:
Sorry, a bit longer explanation is available in http://mathb.in/39134 .

Reply:
Thank you a lot @traversaro, now it makes sense and it's very clear! I pushed new commits that reflect the formulation you linked.